dotcom-rendering/src/model/enhance-placeholders.ts

```diff
@@ -0,0 +1,84 @@
+import { Design } from '@guardian/types';
+import { decideDesign } from '@root/src/web/lib/decideDesign';
+
+export const replacePlaceholders = (
+	elem: InteractiveAtomBlockElement,
+	variables: Map<string, string>,
+): InteractiveAtomBlockElement => {
+	if (!elem.html) {
+		return elem;
+	}
+
+	const replacer = (substring: string, name: string): string => {
+		return variables.get(name) || substring;
+	};
+
+	// Note: use str.replace with 'g' (global) flag as str.replaceAll not
+	// supported in current node version (14.6).
+	const re = /{{\s*(\w+)\s*}}/g;
+	elem.html = elem.html.replace(re, replacer);
+
+	return elem;
+};
+
+/**
+ * generateVariables defines the templated variables we replace. As such, it
+ * represents an ongoing contract with interactive atoms; fields should not be
+ * removed from it once added so be cautious when thinking about adding new
+ * ones.
+ */
+const buildPlaceholders = (data: CAPIType): Map<string, string> => {
+	return new Map(
```

Comment:
out of curiosity, why do we chose a Map here (rather than just a plain Object)? I have tended just to use Objects in similar situations so would be glad to understand why Map is better :)

Reply:
I'm also curious about this!

Reply:
Good question